public/store/DetectorsStore.tsx

```diff
@@ -0,0 +1,329 @@
+
+      let title: string = `Create detector failed.`;
+      if (!mappingsResponse.ok) {
+        const message = 'Double check the field mappings and try again.';
+
+        this.showNotification(
+          title,
+          message,
+          'danger',
+          false,
+          'Review detector configuration',
+          DataStore.detectors.viewDetectorConfiguration
+        );
+
+        return Promise.resolve({
+          ok: false,
+          error: { title, message },
+        });
+      }
+
+      if (!detectorResponse.ok) {
+        this.showNotification(
+          title,
+          detectorResponse.error,
+          'danger',
+          false,
+          'Review detector configuration',
+          DataStore.detectors.viewDetectorConfiguration
+        );
+
+        return Promise.resolve({
+          ok: false,
+          error: {
+            title,
+            message: detectorResponse.error,
+          },
+        });
+      }
+
+      let dashboardId;
+      const detector = detectorResponse.response.detector;
+      const detectorId = detectorResponse.response._id;
+      if (logTypesWithDashboards.has(detector.detector_type)) {
+        const dashboardResponse = await this.createDashboard(
+          detector.name,
+          detector.detector_type,
+          detectorId,
+          detector.inputs[0].detector_input.indices
+        );
+        if (dashboardResponse && dashboardResponse.ok) {
+          dashboardId = dashboardResponse.response.id;
+        } else {
+          const dashboards = await this.savedObjectsService.getDashboards();
+          dashboards.some((dashboard) => {
+            if (dashboard.references.findIndex((reference) => reference.id === detectorId) > -1) {
+              dashboardId = dashboard.id;
+              return true;
+            }
+
+            return false;
+          });
+        }
+      }
+
+      const goToDetectorDetails = (e: any) => {
+        e.preventDefault();
+        DataStore.detectors.deleteState();
+        this.history?.push(`${ROUTES.DETECTOR_DETAILS}/${detectorId}`);
+      };
+
+      title = `Detector created successfully: ${detectorResponse.response.detector.name}`;
+      this.showNotification(
+        title,
+        undefined,
+        'success',
+        false,
+        'View detector',
+        goToDetectorDetails
+      );
+
+      return Promise.resolve({
+        detectorId: detectorId,
+        dashboardId: dashboardId,
+        ok: true,
+      });
+    }
+
+    return Promise.resolve({ ok: false });
+  };
+
+  private createDashboard = (
+    detectorName: string,
+    logType: string,
+    detectorId: string,
+    inputIndices: string[]
+  ) => {
+    return this.savedObjectsService
+      .createSavedObject(detectorName, logType, detectorId, inputIndices)
+      .catch((error: any) => {
+        console.error(error);
+      });
+  };
+
+  private showCallout = (callout?: ICalloutProps | undefined): void => {};
```

Comment:
Methods `showCallout` and `showToast` are not implemented yet, is there plan to do it in follow up PR?

Reply:
there are dummy methods and are set to real methods from the Main component to show/hide callouts and toast messages
perhaps renaming them to calloutCallback and toastCallback would make more sense? 
This assignment happens on the Main component. I did try to get these assigned on the app but got a hooks error when using hooks in the app component. I see these happening if react and react-dom versions do not match, but I don't see updating these as a possibility, right? So I moved that to the Main component.

Reply:
Got it, yeah let's name them as callbacks

Reply:
Done.